### DISCUSSION TITLE
Mermaid integration in Wiki pages

### DISCUSSION THREAD

**Discussion by SebastienFCT** [community/community#11612] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello there,

The [recent integration of mermaid in Github](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) is awesome, Kudos to everyone who worked on this!

I've noticed it hasn't been implemented on the Wiki pages

- Wiki Editor:

<img width="1264" alt="Screen Shot 2022-02-16 at 11 40 55 AM" src="https://user-images.githubusercontent.com/2982335/154313490-28a9188c-70ee-4b9e-9017-ad62ee35c9a7.png">

- Output:

<img width="917" alt="Screen Shot 2022-02-16 at 11 40 46 AM" src="https://user-images.githubusercontent.com/2982335/154313488-bdcd5179-63e1-4fc9-8107-e3f36ab75ced.png">

Any plan of doing so? That'd be awesome to be able to integrate our diagrams in there.

Cheers

**Comment by michellemerrill** [comment]
Also, see duplicate discussion:  https://github.com/github/feedback/discussions/11820

**Comment by davidstaheli** [comment, marked as answer]
Thank you for your patience with the Mermaid wiki support. We plan to support it as soon as possible, though to be transparent it may not be until March or April. It will be sooner if at all possible. Thank you for the upvotes on this discussion.

**Reply by ferrao** [reply]
Thank you for the feedback @davidstaheli ! Hope it will be available for all supported markup languages and not just markdown, keep up the good work!

**Reply by nathan-fiscaletti** [reply]
Is this still on track for this month or next?

**Reply by mitchelsellers** [reply]
Thanks for this information, it would be incredibly valuable

**Reply by Brutalbeard** [reply]
Really lookin forward to it! Thanks y'all!

**Reply by dougreese** [reply]
It would be great if this thread can be updated when wiki support is released, so we can get notified. I rarely look at the blog.

**Reply by ferrao** [reply]
I get heart palpitations over this every day ;) The problem is that not only this is not supported in Wiki pages, this [extension](https://chrome.google.com/webstore/detail/mermaid-diagrams/phfcghedmopjadpojhmmaffjmfiakfil) also stopped working at some point. 

I used to see all my diagrams using it, but now I have a bunch of wiki pages that I need to build locally every time I need to look at them, actually rendering the whole wiki worthless, which was one of the reasons that got me into a paid account. But I have faith on the GitHub team!

**Reply by euthuppan** [reply]
you can maybe use this one @ferrao meanwhile https://chrome.google.com/webstore/detail/markdown-diagrams/pmoglnmodacnbbofbgcagndelmgaclel

still works for us.

**Reply by davidstaheli** [reply]
I'm very sorry, all, but the wiki support has been delayed until GitHub engineers become available to do the work. I'll update this discussion as soon as I have more information or a new estimated date. I apologize if this inconveniences anyone. I can only imagine how nice it would be to have this support for regular use of wikis. Hopefully a browser extension like the one above will help in the meantime.

**Reply by ferrao** [reply]
I appreciate the effort @davidstaheli , but In my specific case I could not wait any longer, so I ended up moving the entire wiki into a separate documentation web site. It was a bit of work to get everything up&running and perfectly integrated, but on the upside, I ended up with a much better solution, considering all the limitations of the wiki.

Anyway, thank you so much for the feedback, it's nice to be kept in the loop in this way!

**Reply by david-wolgemuth** [reply]
> to be transparent it may not be until March or April. It will be sooner if at all possible

Hi!  Also a big fan of this integration.  We just hit May, so I figured I could re-ask.  Are there any further plans for this to be prioritized?

**Reply by roberttravispierce** [reply]
Thank you for supporting Mermaid. We are using it in our team extensively and would also very much like to use it in our Wiki pages also. Since it's May now, could we possibly get an updated timeline? Thanks.

**Reply by RobReeveLexTego** [reply]
Same here - we are moving from Confluence, and were moving our product wiki here. I will now be stopping the migration :-(

We were moving from PlantUML to Mermaid - because it is supported in Github (just not the wiki)...

**Reply by alexkutsan** [reply]
I am also waiting for this changes in github :-)

**Reply by tom-haines** [reply]
Yes we just started a migration from asciidoc, and now realise it isn't supported in the wiki.  Any updates as to eta would be appreciated as whether we abandon migration or can continue with it (depending on whether mermaid support is being added to wiki markdown pages, or not).

**Reply by alexkutsan** [reply]
Are there any other abilities to embed diagrams into the GitHub wiki(planuml or any other?)? 
To avoid links to other resources.

**Comment by rungrotArunon** [comment]
Thanks you bro

**Comment by TarDruggan** [comment]
I would just like to put my 0.02 in that this is a very needed feature for the Wiki. 
The Wiki's value would go up by quite a bit to be able to do diagrams.  Currently doing this in VS Code and checking in diagrams for people to check-out and view as they are reading the Wiki. Not a good solution. 
As soon as this works we will be using it.

**Comment by walbourn** [comment]
I use the GitHub wiki's extensively for my online documentation. Diagraming via Markdown would be a *huge* improvement.

**Comment by gcerik** [comment]
Any updates on this? Mermaidjs is a game changer for documentation

**Reply by KhaaL** [reply]
+1 on updates, was quite suprised to see lack of mermaid support on the wiki's...

**Comment by duanxinhuan** [comment]
Any updates?

**Comment by lc-caigwatkin** [comment]
Any updates?

**Comment by michaelgevaryahu** [comment]
Also curious about this - would love to see this feature.

**Comment by davidstaheli** [comment]
Support for Mermaid in wiki Markdown is likely to release by the end of August!

**Reply by AngelJrz** [reply]
Thanks for this update!

**Comment by rushtonmd** [comment]
Mermaid (and other fun things) are now supported in Wikis 🎉 See the [changelog](https://github.blog/changelog/2022-08-09-wikis-now-support-math-and-mermaid-diagrams/) for more info.

Enjoy! 🙇
